### PR TITLE
asynchronous EmojiProvider.getCandidates processing

### PR DESCRIPTION
**Pull request by philippelaporteproduct**
https://linear.app/productscience/issue/CUS-913

Reportedly these changes no longer improve the situation in the latest code

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java [hunk 1]
```diff
@@ -36,6 +36,25 @@ public class EmojiProvider {
   private static final    String TAG   = Log.tag(EmojiProvider.class);
   private static final    Paint  PAINT = new Paint(Paint.FILTER_BITMAP_FLAG | Paint.ANTI_ALIAS_FLAG);
 
+  public interface onCandidatesLoadedListener {
+    void onCandidatesLoaded(EmojiParser.CandidateList candidates);
+  }
+
+  public static void getCandidates(@Nullable CharSequence text, onCandidatesLoadedListener onCandidatesLoaded) {
```
Comment (briancooley-ps) [inline app/src/main/java/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java:43]: IMO `getCandidates` is a somewhat misleading name as I expect it to return something. Maybe `loadCandidates` to more closely match the `onCandidatesLoaded` method in the interface?

File: app/src/main/java/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java [hunk 2]
```diff
@@ -36,6 +36,25 @@ public class EmojiProvider {
   private static final    String TAG   = Log.tag(EmojiProvider.class);
   private static final    Paint  PAINT = new Paint(Paint.FILTER_BITMAP_FLAG | Paint.ANTI_ALIAS_FLAG);
 
+  public interface onCandidatesLoadedListener {
```
Comment (briancooley-ps) [inline app/src/main/java/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java:39]: Style nit: usually interfaces are capitalized.

**Review by briancooley-ps — APPROVED** [review]
Couple of nits, but nothing to hold up moving forward. Nice work.

**Comment by philippelaporteproduct** [comment]
improvement no longer valid in latest version, closing PR

**Pull request closed without merging**